Libplanet.Net/Consensus/ConsensusReactor.cs

```diff
@@ -127,12 +131,98 @@ private async Task ProcessMessageHandler(Message message)
                     break;
             }
         }
+
+        private async Task RequestBlockNotExistsAsync(ConsensusMessage consensusMessage)
+        {
+            if (!_context.ContainsBlock(consensusMessage.BlockHash))
+            {
+                var hashList = new List<BlockHash>
+                {
+                    consensusMessage.BlockHash,
+                };
+                var message = new GetBlocks(hashList)
+                {
+                    Remote = _transport.AsPeer,
+                };
+
+                _logger.Debug(
+                    "{MethodName}: Requesting Block {BlockHash} derived from {@Message}",
+                    nameof(RequestBlockNotExistsAsync),
+                    consensusMessage.BlockHash,
+                    consensusMessage);
+
+                var blockMessage = await _transport.SendMessageAsync(
+                    _routingTable.GetPeer(consensusMessage.Remote!.Address),
+                    message,
+                    TimeSpan.FromSeconds(1),
+                    CancellationToken.None);
+
+                if (blockMessage is Messages.Blocks)
+                {
+                    var unmarshalBlock = BlockMarshaler.UnmarshalBlock<T>(
+                        _context.HashAlgorithm,
+                        (Bencodex.Types.Dictionary)_codec.Decode(blockMessage.DataFrames.Last())
+                    );
+
+                    _context.PutBlockToStore(unmarshalBlock);
+
+                    _logger.Debug(
+                        "{MethodName}: Received Block {BlockHash} from {Remote}",
+                        nameof(RequestBlockNotExistsAsync),
+                        unmarshalBlock.Hash,
+                        message.Remote);
+
+                    if (_context.IsVoteOnHold)
+                    {
+                        HandleMessage(
+                            new ConsensusVote(_context.CurrentRoundContext.Voting(VoteFlag.Absent)));
+                    }
+                }
+            }
+        }
+
+        private async Task ResponseBlockAsync(GetBlocks message)
+        {
+            var block = _context.GetBlockFromStore(message.BlockHashes.Last());
+            if (block != null)
+            {
+                var listBlock = new List<byte[]>
+                {
+                    _codec.Encode(block.MarshalBlock()),
+                };
+
+                var sending = new Messages.Blocks(listBlock)
+                {
+                    Identity = message.Identity,
+                    Remote = _transport.AsPeer,
+                };
+
+                _logger.Debug(
+                    "{MethodName}: Received Block request {BlockHash} from {Remote}",
+                    nameof(RequestBlockNotExistsAsync),
+                    message.BlockHashes.First(),
+                    message.Remote);
+
+                await _transport.ReplyMessageAsync(sending, CancellationToken.None);
+            }
+
+            await ReplyPongAsync(message);
```

Comment:
I don't think this is required because a reply `sending` already sent? I guess you should make an `else` code block and return `Pong`?